core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -67,9 +67,7 @@ object StaticFile {
 
   def fromFile(f: File, start: Long, end: Long, buffsize: Int, req: Option[Request])
                         (implicit es: ExecutorService): Option[Response] = {
-
-    if (start < 0 || end < start || buffsize <= 0)
-      throw new Exception(s"start: $start, end: $end, buffsize: $buffsize")
+    require (start >= 0 && end > start && buffsize > 0, s"start: $start, end: $end, buffsize: $buffsize")
```

Comment:
I don't like exceptions either. However, I think ```Some(InternalServerError)``` is not the solution for two main reasons:
* Unless they are explicitly logging 5xx's (which they should...) the failure will be silent and even then they won't get a descriptive message making it hard to figure out they are not checking the arguments to StaticFile. 
* If the route wants to attempt to recover it has to begin peeking at the status code which is never going to be intuitive or worth it.

This situation reminds me of places like https://github.com/scalaz/scalaz-stream/blob/292230aaa158bb4141da2a1aa15c23df818209df/src/main/scala/scalaz/stream/Process.scala#L845 (among others) where you should be verifying the arguments before calling the function.
Maybe there is another way to deal with this?